VoiceProcessor.swift

```diff
@@ -15,20 +15,28 @@ public class VoiceProcessor {
     private let numBuffers = 3
     private var audioQueue: AudioQueueRef?
     private var audioCallback: (([Int16]) -> Void)?
+    private var sessionErrorCallback: ((Error) -> Void)?
     private var frameLength: UInt32?
+    private var bufferRef: AudioQueueBufferRef?
     
     private var started = false
+    private let session: AVAudioSession
     
-    private init() {}
+    private init() {
+        self.session = AVAudioSession.sharedInstance()
```

Comment:
do we need to hold on to AVAudioSession? It might be better to call the sharedInstance() whenever we need it - I'm not sure if this reference could change, but we might as well be extra safe.